lib/apr/events.ex

```diff
@@ -74,50 +83,88 @@ defmodule Apr.Events do
     |> Repo.insert()
   end
 
-  @doc """
-  Updates a event.
-
-  ## Examples
-
-      iex> update_event(event, %{field: new_value})
-      {:ok, %Event{}}
-
-      iex> update_event(event, %{field: bad_value})
-      {:error, %Ecto.Changeset{}}
-
-  """
-  def update_event(%Event{} = event, attrs) do
-    event
-    |> Event.changeset(attrs)
-    |> Repo.update()
+  def get_order_events(status \\ "approved", day_threshold \\ 1) do
+    with events <- list_events(routing_key: "order.#{status}", day_threshold: day_threshold),
+         {:ok, artworks} <- fetch_artworks(events) do
+      {:ok, %{events: events, artworks: artworks, totals: get_totals(events)}}
+    else
+      # we can do lot better here, i think actual solution is to have get events return {:ok,...} {:error, ..} instead of doing assign
+      [] -> {:ok, events: [], totals: %{amount_cents: 0, commission_cents: 0}, artworks: %{}}
+      {:error, error} -> {:error, error}
+    end
   end
 
-  @doc """
-  Deletes a Event.
 
-  ## Examples
-
-      iex> delete_event(event)
-      {:ok, %Event{}}
-
-      iex> delete_event(event)
-      {:error, %Ecto.Changeset{}}
-
-  """
-  def delete_event(%Event{} = event) do
-    Repo.delete(event)
+  defp fetch_artworks([]), do: []
+
+  defp fetch_artworks(order_events) do
+    Neuron.Config.set(url: Application.get_env(:apr, :metaphysics)[:url])
+
+    order_id_artwork_ids =
+      order_events
+      |> Enum.reduce(%{}, fn e, acc ->
+        artwork_ids =
+          e.payload["properties"]["line_items"] |> Enum.map(fn li -> li["artwork_id"] end)
+
+        acc
+        |> Map.merge(%{e.payload["object"]["id"] => artwork_ids})
+      end)
+
+    uniq_artwork_ids = order_id_artwork_ids |> Map.values() |> List.flatten() |> Enum.uniq()
+
+    fetch_response =
+      Neuron.query(
+        """
+          query orderArtworks($ids: [String]) {
+            artworks(ids: $ids) {
+              id
+              _id
+              title
+              artist {
+                name
+              }
+              partner {
+                name
+              }
+              imageUrl
+            }
+          }
+        """,
+        %{ids: uniq_artwork_ids}
+      )
+
+    case fetch_response do
+      {:ok, response} ->
+        artworks_map =
+          response.body["data"]["artworks"]
+          |> Enum.reduce(%{}, fn a, acc -> Map.merge(acc, %{a["_id"] => a}) end)
```

Comment:
I'm having a really hard time visualizing what this whole section is doing.

Reply:
yes, and this may not be the best way to do this, basically what happens in here is, we create a map where each key is `artwork[_id]` and each value is the artwork details for that id. then we pass this to the template as `@artworks` and when iterating through events, we try to find the artwork id of that order and then using this map, show the artwork details.